server.py

```diff
@@ -113,5 +129,17 @@ def get_verified_email(jwt):
     return {'email': payload['sub']}
 
 
-host = 'localhost' if 'localhost' in META['RP_ORIGIN'] else '0.0.0.0'
-run(server='gunicorn', host=host, port=int(getenv('PORT', '8080')))
+HANDLERS = {'index': index, 'login': login, 'static': static}
+STATUS = {200: '200 OK', 400: '400 Bad Request'}
+
+def application(env, respond):
+    pi = [] if not env['PATH_INFO'] else env['PATH_INFO'].strip('/').split('/')
+    handler = pi[0] if pi and pi[0] else 'index'
+    status, headers, content = HANDLERS[handler](env)
+    respond(STATUS[status], list(headers.items()))
+    return [content]
+
+
+if __name__ == '__main__':
+    host, port = parse.urlparse(META['RP_ORIGIN']).netloc.split(':')
+    simple_server.make_server(host, int(port), application).serve_forever()
```

Comment:
So with my changes, `server.application` is a valid WSGI app. It should be very easy to wrap another server around it. In fact, that's how I was running it before, with mod_wsgi (but my mod_wsgi is setup to use Python 2.7). So I think that would be a good way to address the performance concerns.

As for legibility and re-usability: I think it's all about exposing people to the core of the Let's Auth code. To me, that's basically the login form and the login POST handler (including `get_verified_email()`). The rest of it is pretty much fluff. So I think it's easier to grok the code if we don't have session handling.

As for re-use, I do think it can be quite powerful to make it easy for people to standup their own demo RP, or to copy/paste some bits into their app to make their own RP. The more opinionated the demo RP is, the harder it is to do that.